src/renderer/components/MainPage.tsx

```diff
@@ -292,20 +316,22 @@ class MainPage extends React.PureComponent<Props, State> {
         if (addedIndex === undefined || removedIndex === addedIndex) {
             return;
         }
-        if (!this.state.activeServerName) {
-            return;
-        }
-        const currentTabs = this.props.teams.find((team) => team.name === this.state.activeServerName)?.tabs;
-        if (!currentTabs) {
+        if (!(this.state.activeServerId && this.state.tabs.has(this.state.activeServerId))) {
             // TODO: figure out something here
             return;
         }
-        const teamIndex = this.props.moveTabs(this.state.activeServerName, removedIndex, addedIndex < currentTabs.length ? addedIndex : currentTabs.length - 1);
-        if (!teamIndex) {
-            return;
-        }
-        const name = currentTabs[teamIndex].name;
-        this.handleSelectTab(name);
+        const currentTabs = this.state.tabs.get(this.state.activeServerId)!;
+        const tabsCopy = currentTabs.concat();
```

Comment:
Would we expect to need a deep copy here because we might modify the individual tabs at some point down the line?